packages/sdk/src/actions/getDepositFromLogs.ts

```diff
@@ -14,47 +15,94 @@ export type GetDepositLogsParams = {
   }>;
 };
 
-export type GetDepositLogsReturnType = ReturnType<typeof getDepositFromLogs>;
+export type GetDepositLogsReturnType = ReturnType<typeof parseDepositLogs>;
 
-export function getDepositFromLogs({
-  originChainId,
-  receipt,
-  filter,
-}: GetDepositLogsParams): Deposit | undefined {
-  const parsedLogs = parseEventLogs({
-    abi: spokePoolAbi,
-    eventName: "V3FundsDeposited",
-    logs: receipt.logs,
+export function parseDepositLogs(
+  logs: Log[],
+  filter?: Partial<{
+    inputToken: Address;
+    outputToken: Address;
+    destinationChainId: bigint;
+    inputAmount: bigint;
+    outputAmount: bigint;
+  }>,
+): DepositLog | undefined {
+  const blockData = {
+    depositTxHash: logs[0]!.blockHash!,
+    depositTxBlock: logs[0]!.blockNumber!,
+  };
+  // Parse V4 Logs
+  const parsedV4Logs = parseEventLogs({
+    abi: spokePoolAbiV4,
+    eventName: "FundsDeposited",
+    logs,
     args: filter,
   });
 
-  if (parsedLogs.length > 1) {
-    throw new Error("Multiple deposit logs found. Specify filter");
+  const v4Log = parsedV4Logs?.[0];
```

Comment:
As mentioned somewhere above: We should probably remove the v4 naming here as this upgrade is technically not a major one. So that in the future when the real v4 upgrade comes, we don't have a naming collision